ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveStorageHandler.java

```diff
@@ -550,14 +550,19 @@ default boolean isTimeTravelAllowed() {
     return false;
   }
 
-  default boolean isMetadataTableSupported() {
+  default boolean isTableMetaRefSupported() {
     return false;
   }
```

Comment:
Isn't this an incompatible change?

Reply:
Sure. It is mainly used to check table if having metadata table or branch&tag. Acutually, we think it is only used by iceberg.

Reply:
I think we shouldn’t assume, but maintain compat and keep the name as is

Reply:
ha, i am not obect to keep the method as it is. Let me give some thought.

The `isMetadataTableSupported()` is only used here https://github.com/apache/hive/pull/4292/files#diff-9d6bc2f1a2972d844b3107f29bc1731af7adf6f4c922e08904345c5ae88cd727L1755, and then `storagehandler::isValidMetadataTable` will check the metadata table if is valid.

That is to say, the actual effect of method  `isMetadataTableSupported()`  is just simply to verify that the table has three layer identifier `database.table.tableRef`, and the final check occurs in `storagehandler::isValidMetadataTable`.  And i rename it from `isMetadataTableSupported()` to `isTableMetaRefSupported()` to make its three_layer_identifier meaning more general, not just for metadata table.

But maybe i am not good at naming & renaming method. If you think we should keep hte name as is or using another method name, please feel free to give advice. Thanks.

Reply:
Should keep the name as is IMO

Reply:
> Sure. It is mainly used to check table if having metadata table or branch&tag. Acutually, we think it is only used by iceberg.

it was only introduced for iceberg, so i don't think it should be an issue to alter the name

Reply:
We don't just have it in Iceberg code:
https://github.com/apache/hive/blob/rel/release-4.0.0-alpha-2/ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveStorageHandler.java#L464-L465

We added in the parent interface itself and it is indeed released in the previous release also.

https://github.com/apache/hive/blob/rel/release-4.0.0-alpha-2/ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveStorageHandler.java#L464-L465

So, technically if some client code does getStorageHandler() and calls this it is gonna fail, or if someone had a custom/internal storage handler where he implemented it, he is gonna suffer as well.

Mostly nobody would have, but we can't gurantee based on your sentiments :-), But it is still an incompatible change. "Changing a public method name/signature", I think this method name change ain't serving any functional benefits, we can keep it as is


If people still want it: The correct way is deprecate the original, add a new one, call the new one from the deprecated method. But I don't think we want it that hard. I would be against breaking compat just for look & feel :-)

Reply:
Thanks @ayushtkn for explanation. Sound reasonable. IMO, maybe we should keep the name as is  `isMetadataTableSupported()` and add a new method `isTableMetaRefSupported()`?  wdyt, @deniskuzZ

Reply:
sounds reasonable, however, please add Javadoc for both, and maybe mark isMetadataTableSupported() as deprecated so we could clean the interface in the next releases.

Reply:
Done. Please take a look again if my doc description is reasonable. Thanks all.